Modules/Scripted/SegmentStatistics/SegmentStatisticsCalculators/SegmentStatisticsCalculatorBase.py

```diff
@@ -0,0 +1,201 @@
+import vtk
+import qt
+import slicer
+
+
+class SegmentStatisticsCalculatorBase(object):
+  """Base class for statistics calculators operating on segments.
+  Derived classes should specify: self.name, self.keys, self.defaultKeys
+  and implement: computeStatistics, getMeasurementInfo
+  """
+
+  @staticmethod
+  def initCodedEntry(codeValue, codingScheme, codeMeaning, returnAsString=False):
+    entry = slicer.vtkCodedEntry()
+    entry.SetValueSchemeMeaning(codeValue, codingScheme, codeMeaning)
+    return entry if not returnAsString else entry.GetAsString()
+
+  @staticmethod
+  def getDICOMTriplet(codeValue, codingSchemeDesignator, codeMeaning):
+    """Utility method to form DICOM triplets in getMeasurementInfo"""
+    return {'CodeValue':codeValue,
+            'CodingSchemeDesignator': codingSchemeDesignator,
+            'CodeMeaning': codeMeaning}
+
+  @staticmethod
+  def generateMeasurementInfo(name, description, units, quantityCode, unitsCode, measurementMethodCode=None,
+                              derivationCode=None):
+    info = {
+      "name": name,
+      "description": description,
+      "units": units,
+      "DICOM.QuantityCode": quantityCode,
+      "DICOM.UnitsCode": unitsCode
+      }
+    if measurementMethodCode:
+      info["DICOM.MeasurementMethodCode"] = measurementMethodCode
+    if derivationCode:
+      info["DICOM.DerivationCode"] = derivationCode
+    return info
+
+  def __init__(self):
+    self.name = "" # name of the statistics calculator
+    self.keys = () # keys for all supported measurements; should have format "CalculatorName.measurement"; e.g. "Labelmap.volume cc"
+    self.defaultKeys = () # measurements that will be enabled by default
+    self.requestedKeysCheckboxes = {}
+    self.parameterNode = None
+    self.parameterNodeObserver = None
+
+  def __del__(self):
+    if self.parameterNode and self.parameterNodeObserver:
+      self.parameterNode.RemoveObserver(self.parameterNodeObserver)
+
+  def computeStatistics(self, segmentID):
+    """Compute measurements for requested keys on the given segment"""
+    stats = {}
+    #segmentationNode = slicer.mrmlScene.GetNodeByID(self.getParameterNode().GetParameter("Segmentation"))
```

Comment:
I removed it. I had it as a reference for plugin developers but the plugins themselves are probably better references.